### PR TITLE
Avoid containment-related DeprecationWarning on Python 3.7

### LINKED ISSUE #853: DeprecationWarning from test suite on Python 3.7
Under Python 3.7, the test suite (run with `python -m unittest discover traits`) gives the following warning:
```
/Users/mdickinson/.venvs/traits-py37/lib/python3.7/site-packages/traits/trait_base.py:209: DeprecationWarning: using non-Enums in containment checks will raise TypeError in Python 3.8
```

### PR DESCRIPTION
**Pull request by mdickinson**
Also adds some unit tests for `safe_contains`.

Fixes #853

### CODE REVIEW

File: traits/trait_base.py [hunk 1]
```diff
@@ -205,6 +205,11 @@ def safe_contains(value, container):
     TypeError.  In these cases we make the (reasonable) assumption that the
     value is _not_ contained in the container.
     """
+    # Do a LBYL check for Enums, to avoid the DeprecationWarning issued
+    # by Python 3.7. Ref: enthought/traits#853.
+    if isinstance(container, enum.EnumMeta) and not isinstance(value, enum.Enum):
```
Comment (mdickinson) [inline traits/trait_base.py]: I'm not hugely happy with this solution. An alternative is to use the `filterwarnings` machinery to explicitly filter out the relevant `DeprecationWarning`, but that's awkward, too: either you do it globally, in which case you're violating DNMWGSTDNBTY, or you have the performance hit of saving and restoring warnings state on every call to `safe_contains`.

Comment (corranwebster) [reply traits/trait_base.py]: What happens if you have an `enum.Enum` instance that is not an enum from `container`?  Do we need `not isinstance(value, container)` instead?

Comment (mdickinson) [reply traits/trait_base.py]: I think that's covered by the test suite.

Comment (mdickinson) [reply traits/trait_base.py]: Actually, I'm not sure I understand what you're asking. Are you asking about the case that `value` is an `enum.Enum` instance and `container` is a different `enum.Enum` class?

Comment (corranwebster) [reply traits/trait_base.py]: Yes - question resolved in offline conversation.

**Pull request merged**